addons/account/models/account_account.py

```diff
@@ -857,11 +874,22 @@ def _onchange_name(self):
             self.name = name
             self.code = code
 
-    @api.depends_context('company')
+    @api.depends_context('company', 'formatted_display_name')
     @api.depends('code')
     def _compute_display_name(self):
+        formatted_display_name = self.env.context.get('formatted_display_name')
+        preferred_account_ids = self.env.context.get('preferred_account_ids', [])
+        if (
+            (move_type := self.env.context.get('move_type'))
+            and (partner := self.env.context.get('partner_id'))
+            and not preferred_account_ids
+        ):
+            preferred_account_ids = self._order_accounts_by_frequency_for_partner(self.env.company.id, partner, move_type)
         for account in self:
-            account.display_name = f"{account.code} {account.name}" if account.code else account.name
+            if formatted_display_name and account.code:
+                account.display_name = f"{account.code} {account.name} {_('`Suggested`') if account.id in preferred_account_ids else ''}{f'<br>--{account.description}--' if account.description else ''}"
```

Comment:
Keep the formatting out of the translation
```suggestion
                account.display_name = f"""{account.code} {account.name}{f' `{_("Suggested")}`' if account.id in preferred_account_ids else ''}{f'<br>--{account.description}--' if account.description else ''}"""
```

Could also spread multiple lines
```suggestion
                account.display_name = (
                    f"""{account.code} {account.name}"""
                    f"""{f' `{_("Suggested")}`' if account.id in preferred_account_ids else ''}"""
                    f"""{f'<br>--{account.description}--' if account.description else ''}"""
                )
```